microsetta_interface/implementation.py

```diff
@@ -813,25 +812,71 @@ def get_create_human_source(*, account_id=None):
     if has_error:
         return consent_output
 
+    form_type = "source"
+
     return _render_with_defaults(
         'new_participant.jinja2',
         tl=consent_output,
         post_url=post_url,
         duplicate_email_check_url=duplicate_email_check_url,
         home_url=home_url,
+        form_type=form_type,
         language_tag=session_locale())
 
 
 @prerequisite([ACCT_PREREQS_MET])
 def post_create_human_source(*, account_id=None, body=None):
-    has_error, consent_output, _ = ApiRequest.post(
-        "/accounts/{0}/consent".format(account_id), json=body)
-    if has_error:
-        return consent_output
 
-    new_source_id = consent_output["source_id"]
+    create_new_source = False
+    consent_type = body.get("consent_type")
+
+    if "Data" in consent_type:
+        consent_type = "Data"
 
-    return _refresh_state_and_route_to_sink(account_id, new_source_id)
+        if "source_id" in session:
+            source_id = session['source_id']
+
+            has_error, consent_output, _ = ApiRequest.post(
```

Comment:
What happens is there's an error signing the consent document? There's no condition for if has_error here.

Reply:
Added check to handle has_error.